src/controller/src/lib.rs

```diff
@@ -742,15 +714,8 @@ where
             immediate_watch_sets: Vec::new(),
         };
 
-        // We have some logic that we want to run both when initialized with
-        // `read_only = false` _and_ when later transitioning out of read-only
-        // mode. This way we keep that logic in one place.
-        if !read_only {
-            // We did not yet tell the controller about any collections, so
-            // there are no tables to initialize yet. Hence we don't need a
-            // register_ts.
-            let register_ts = None;
-            this.allow_writes(register_ts).await;
+        if !this.read_only {
+            this.remove_past_generation_replicas_in_background();
         }
```

Comment:
Nit: We could move this part before the creation of `Self` and then get rid of the `this` variable.

Reply:
Well it's a method on `Self` 😅. I could inline the method in this method if you prefer though?

Reply:
🤦 Yeah, that's a pretty good reason. No, I prefer the current version over inlining!